jerry-core/include/jerryscript-core.h

```diff
@@ -165,36 +165,31 @@ typedef uint32_t jerry_length_t;
 typedef uint32_t jerry_value_t;
 
 /**
- * Description of ECMA property descriptor.
+ * Status flags of ECMA property descriptor.
  */
-typedef struct
+typedef enum
 {
-  /** Is [[Value]] defined? */
-  bool is_value_defined;
-
-  /** Is [[Get]] defined? */
-  bool is_get_defined;
-
-  /** Is [[Set]] defined? */
-  bool is_set_defined;
-
-  /** Is [[Writable]] defined? */
-  bool is_writable_defined;
-
-  /** [[Writable]] */
-  bool is_writable;
+  JERRY_PROP_NO_OPTS = (0), /** empty property descriptor */
+  JERRY_PROP_IS_GET_DEFINED = (1 << 0), /** Is [[Get]] defined? */
+  JERRY_PROP_IS_SET_DEFINED = (1 << 1), /** Is [[Set]] defined? */
 
-  /** Is [[Enumerable]] defined? */
-  bool is_enumerable_defined;
+  JERRY_PROP_IS_CONFIGURABLE = (1 << 2), /** [[Configurable]] */
+  JERRY_PROP_IS_ENUMERABLE = (1 << 3), /** [[Enumerable]] */
+  JERRY_PROP_IS_WRITABLE = (1 << 4), /** [[Writable]] */
 
-  /** [[Enumerable]] */
-  bool is_enumerable;
+  JERRY_PROP_IS_VALUE_DEFINED = (1 << 6), /** Is [[Value]] defined? */
+  JERRY_PROP_IS_CONFIGURABLE_DEFINED = (1 << 7), /** Is [[Configurable]] defined? */
+  JERRY_PROP_IS_ENUMERABLE_DEFINED = (1 << 8), /** Is [[Enumerable]] defined? */
+  JERRY_PROP_IS_WRITABLE_DEFINED = (1 << 9), /** Is [[Writable]] defined? */
+} jerry_property_descriptor_status_flags_t;
```

Comment:
What if we use the  `jerry_property_descriptor_flags_t`? The `status` does not provide extra info IMHO.

Reply:
I named it like this in order to follow `ecma_property_descriptor_status_flags_t`, but of course we can name it like you suggest.